src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -0,0 +1,69 @@
+package frc.robot.subsystems;
+
+import java.util.function.BooleanSupplier;
+
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+import com.chopshop166.chopshoplib.motors.SmartMotorController;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.maps.RobotMap.TelescopeMap;
+
+public class Climber extends SmartSubsystemBase {
+
+  // Constants:
+  private final double EXTEND_SPEED = 1.0;
+  private final double RETRACT_SPEED = -1.0;
+
+  private final SmartMotorController motor;
+  private final BooleanSupplier upperLimit;
+  private final BooleanSupplier lowerLimit;
+
+  public Climber(TelescopeMap map) {
+    motor = map.getMotor();
+    upperLimit = map.getUpperLimit();
+    lowerLimit = map.getLowerLimit();
+  }
+
+  public CommandBase extend() {
```

Comment:
These should also use the `motors.Modifier.upperLimit` and `motors.Modifier.lowerLimit` to ensure that the motors don't try to drive against the limit switches. This should solve the issue we talked about where the command will run briefly before checking `isFinished`.

Reply:
See the code here https://github.com/chopshop-166/chopshoplib/blob/main/core/src/main/java/com/chopshop166/chopshoplib/motors/Modifier.java

Reply:
You could probably move the modifiers and group into the class itself so you're not creating a new one every time